common/vcenter_manage_domain_user.yml

```diff
@@ -0,0 +1,121 @@
+#   vcenter_domain_user_first_name: The first name of domain user. Default is same as user name.
+#   vcenter_domain_user_last_name: The last name of domain user. Default is same as domain name.
+#   vcenter_domain_user_password: The domain user password.
+#   vcenter_domain_user_group: The user group which domain user belongs to. Default is empty.
+#   vcenter_domain_user_op: The operation of managing domain user: add or delete.
+#
+- name: "Check vCenter Server is defined and added into memory inventory"
+  ansible.builtin.assert:
+    that:
+      - vcenter_is_defined is defined
+      - vcenter_is_defined
+      - groups['vcenter'] is defined
+      - vcenter_hostname in groups['vcenter']
+    fail_msg: >-
+      vCenter Server information and its SSH username and password must be set in testing vars file.
+      Current vcenter_is_defined is {{ vcenter_is_defined | default('undefined') }},
+      and vCenter Server hostname in memory inventory is
+      {{ groups['vcenter'] is defined and vcenter_hostname in groups['vcenter'] }}.
+
+- name: "Check vcenter_domain_user_name and vcenter_domain_user_op is set"
+  ansible.builtin.assert:
+    that:
+      - vcenter_domain_user_name is defined
+      - vcenter_domain_user_name
+      - vcenter_domain_user_op is defined
+      - vcenter_domain_user_op in ['add', 'delete']
+    fail_msg: >-
+      Parameter 'vcenter_domain_user_name' or 'vcenter_domain_user_op' is incorrect
+      for managing domain user.
+      Current vcenter_domain_user_name is {{ vcenter_domain_user_name | default('undefined') }},
+      and vcenter_domain_user_op is {{ vcenter_domain_user_op | default('undefined') }}.
+
+- name: "Set default user domain of vCenter Server"
+  ansible.builtin.set_fact:
+    vcenter_domain_name: "{{ vcenter_username.split('@')[-1] }}"
+  when: vcenter_domain_name is undefined or not vcenter_domain_name
+
+- name: "Set command for adding domain user"
+  ansible.builtin.set_fact:
+    manage_domain_user_cmd: >-
+      /usr/lib/vmware-vmafd/bin/dir-cli user create
+      --account "{{ vcenter_domain_user_name }}"
+      --user-password "{{ vcenter_domain_user_password }}"
+      --first-name "{{ vcenter_domain_user_first_name | default(vcenter_domain_user_name) }}"
+      --last-name "{{ vcenter_domain_user_last_name | default(vcenter_domain_name) }}"
+      --login "{{ vcenter_username }}"
+      --password "{{ vcenter_password }}"
+  when: vcenter_domain_user_op == "add"
+
+- name: "Set command for deleteing domain user"
+  ansible.builtin.set_fact:
+    manage_domain_user_cmd: >-
+      /usr/lib/vmware-vmafd/bin/dir-cli user delete
+      --account "{{ vcenter_domain_user_name }}"
+      --login "{{ vcenter_username }}"
+      --password "{{ vcenter_password }}"
+  when: vcenter_domain_user_op == "delete"
+
+- name: "{{ vcenter_domain_user_op | capitalize }} domain user '{{ vcenter_domain_user_name }}@{{ vcenter_domain_name }}'"
+  ansible.builtin.command: "{{  manage_domain_user_cmd }}"
```

Comment:
{{  manage_domain_u, it seems there are two Blanks here.

Reply:
Fixed.